src/components/OptionRow.js

```diff
@@ -75,176 +76,202 @@ const defaultProps = {
-                                            size={CONST.AVATAR_SIZE.DEFAULT}
-                                            secondAvatarStyle={[
-                                                props.optionIsFocused
-                                                    ? StyleUtils.getBackgroundAndBorderStyle(focusedBackgroundColor)
-                                                    : undefined,
-                                                hovered && !props.optionIsFocused
-                                                    ? StyleUtils.getBackgroundAndBorderStyle(hoveredBackgroundColor)
-                                                    : undefined,
-                                            ]}
-                                            avatarTooltips={props.option.isPolicyExpenseChat ? [props.option.subtitle] : avatarTooltips}
-                                        />
-                                    )
-                                )
+                <Hoverable
+                    containerStyles={[
+                        this.props.isDisabled ? styles.userSelectNone : null,
+                    ]}
+                >
+                    {hovered => (
+                        <TouchableOpacity
+                            ref={el => touchableRef = el}
+                            onPress={(e) => {
+                                this.setState({isDisabled: true});
+                                if (e) {
+                                    e.preventDefault();
+                                }
+                                let result = this.props.onSelectRow(this.props.option, touchableRef);
+                                if (!(result instanceof Promise)) {
+                                    result = Promise.resolve();
                                 }
-                                <View style={contentContainerStyles}>
-                                    <DisplayNames
-                                        accessibilityLabel="Chat user display names"
-                                        fullTitle={props.option.text}
-                                        displayNamesWithTooltips={displayNamesWithTooltips}
-                                        tooltipEnabled={props.showTitleTooltip}
-                                        numberOfLines={1}
-                                        textStyles={displayNameStyle}
-                                        shouldUseFullTitle={props.option.isChatRoom || props.option.isPolicyExpenseChat}
-                                    />
-                                    {props.option.alternateText ? (
-                                        <Text
-                                            style={alternateTextStyle}
+                                InteractionManager.runAfterInteractions(() => {
+                                    result.then(() => this.setState({isDisabled: this.props.isDisabled}));
+                                });
```

Comment:
The callback is evaluated before the promise is resolved. This caused a regression (after we copied the logic). Please check https://github.com/Expensify/App/pull/18122#discussion_r1246506016 for details.